src/main.rs

```diff
@@ -187,6 +188,27 @@ Run Seafowl with --one-off instead to run a one-off command from the CLI."
         exit(-1);
     }
 
+    // Add a GC task for purging obsolete objects from the catalog and the store
+    if config.misc.gc_interval > 0 {
+        let mut shutdown_r = shutdown.subscribe();
+        let mut interval =
+            interval(Duration::from_secs((config.misc.gc_interval * 3600) as u64));
+        tasks.push(
+            async move {
+                loop {
+                    tokio::select! {
+                        _ = interval.tick() => gc_partitions(&context).await,
```

Comment:
Can `gc_partitions` panic for whatever reason and in that case, will this also crash the whole program? I'm not sure what the right thing to do here is. We could ignore the panic and keep going (log and GC on the next run, similarly to how the HTTP frontend does it if a request handler panics) or crash.

Reply:
Not sure whether it can panic or not, but if it does it crashes the whole program (just tried it out).

I could wrap it up with: https://doc.rust-lang.org/std/panic/fn.catch_unwind.html?